### DISCUSSION TITLE
Unable to access the original Runnable object in SpringSchedulingRunnableWrapper

### DISCUSSION THREAD

**Discussion by kes2464** [open-telemetry/opentelemetry-java-instrumentation#8657] [Q&A] [answered]
Hi

```
if (scheduledTask instanceof ScheduledMethodRunnable) {
  // do something
}
```

One of my applications have the above code which checks a type of Runnable instance and does something.
It's scheduled through Spring Task scheduler, so it's auto-instrumented. 

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/a2fd908d47e638aa85fc4ce3666c1b29e0258b33/instrumentation/spring/spring-scheduling-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/scheduling/v3_1/TaskSchedulerInstrumentation.java#L47-L52

However, TaskSchedulerInstrumentation wraps the object and replaces the argument, so the condition is never true.

There's also no getter of the original instance in the wrapper.

Is there any common practice/workaround that people use apart from using reflection?

**Comment by trask** [comment, marked as answer]
@kes2464 can you open a bug for this? the Java agent generally tries to avoid impacting application behavior, so this would be good for us to look into.
